File: studio-idea-plugin/src/main/java/com/evolveum/midpoint/studio/ui/connector/generator/ConnectorGeneratorContinueWizard.java
package com.evolveum.midpoint.studio.ui.connector.generator;

import com.evolveum.midpoint.studio.impl.MidPointClient;
import com.evolveum.midpoint.studio.ui.connector.generator.component.GenerateConnectorBadge;
import com.evolveum.midpoint.studio.ui.connector.generator.step.InitialContinueConnectorDevelopmentStep;
import com.evolveum.midpoint.studio.ui.connector.generator.step.connection.*;
import com.evolveum.midpoint.studio.ui.connector.generator.step.objectclass.*;
import com.evolveum.midpoint.xml.ns._public.common.common_3.ConnectorDevelopmentType;
import com.intellij.ui.components.JBList;
import org.jetbrains.annotations.NotNull;

public class ConnectorGeneratorContinueWizard extends ConnectorGeneratorWizard {

    private final MidPointClient client;
    private final ConnectorGeneratorDataModel dataModel = new ConnectorGeneratorDataModel();

    public ConnectorGeneratorContinueWizard(
            @NotNull MidPointClient client,
            @NotNull ConnectorDevelopmentType connectorDevelopmentType
    ) {
        super(client.getProject());
        this.client = client;
        dataModel.connectorDevelopmentType = connectorDevelopmentType;
        getHelpButton().setVisible(false);
        setSize(1300, 900);
        buildSteps();
        init();
    }

    @Override
    protected void buildSteps() {
        myWizardStepsList.clear();

//        myWizardStepsList.add(new InitialContinueConnectorDevelopmentStep(this, client, dataModel, GenerateConnectorBadge.State.IN_PROGRESS, true));
        myWizardStepsList.add(new BaseUrlSpecificationStep(this, client, dataModel, GenerateConnectorBadge.State.IN_PROGRESS, false));
        myWizardStepsList.add(new AuthMethodSupportStep(this, client, dataModel, GenerateConnectorBadge.State.NONE, false));
        myWizardStepsList.add(new AuthScriptsConnectorStep(this, client, dataModel, GenerateConnectorBadge.State.NONE, false));
        myWizardStepsList.add(new CredentialsConnectorStep(this, client, dataModel, GenerateConnectorBadge.State.NONE, false));
        myWizardStepsList.add(new TestConnectionStep(this, client, dataModel, GenerateConnectorBadge.State.NONE, false));

//        myWizardStepsList.add(new ObjectClassesStep(this, client, dataModel, GenerateConnectorBadge.State.NONE, true));
//        myWizardStepsList.add(new SchemaScriptValidationStep(this, client, dataModel, GenerateConnectorBadge.State.NONE, false));
//        myWizardStepsList.add(new SchemaValidationStep(this, client, dataModel, GenerateConnectorBadge.State.NONE, false));
//        myWizardStepsList.add(new SearchEndpointsStep(this, client, dataModel, GenerateConnectorBadge.State.NONE, false));
//        myWizardStepsList.add(new SearchAllScriptValidationStep(this, client, dataModel, GenerateConnectorBadge.State.NONE, false));
//        myWizardStepsList.add(new SearchResultStep(this, client, dataModel, GenerateConnectorBadge.State.NONE, false));

        myWizardStepsList.forEach(this::addStep);
        stepNavigationItems = new JBList<>(visibleListModel);

        updateNavigationMenuByLiveStates();
    }

    @Override
    protected void doOKAction() {
        // TODO what at the end?
        super.doOKAction();
    }
}

File: studio-idea-plugin/src/main/kotlin/com/evolveum/midpoint/studio/ui/connector/generator/ConnectorGeneratorContinueWizard.kt
/*
 * Copyright (C) 2010-2026 Evolveum and contributors
 *
 * Licensed under the EUPL-1.2 or later.
 */

package com.evolveum.midpoint.studio.ui.connector.generator

import com.evolveum.midpoint.studio.impl.MidPointClient
import com.evolveum.midpoint.studio.ui.connector.generator.component.GenerateConnectorBadge
import com.evolveum.midpoint.studio.ui.connector.generator.dataModel.ConnectorGeneratorDataModel
import com.evolveum.midpoint.studio.ui.connector.generator.step.InitialContinueConnectorDevelopmentStep
import com.evolveum.midpoint.studio.ui.connector.generator.step.connection.AuthMethodSupportStep
import com.evolveum.midpoint.studio.ui.connector.generator.step.connection.AuthScriptsConnectorStep
import com.evolveum.midpoint.studio.ui.connector.generator.step.connection.BaseUrlSpecificationStep
import com.evolveum.midpoint.studio.ui.connector.generator.step.connection.CredentialsConnectorStep
import com.evolveum.midpoint.studio.ui.connector.generator.step.connection.TestConnectionStep
import com.evolveum.midpoint.xml.ns._public.common.common_3.ConnectorDevelopmentType

class ConnectorGeneratorContinueWizard(
    private val client: MidPointClient,
    connectorDevelopmentType: ConnectorDevelopmentType
): ConnectorGeneratorWizard(client.project) {

    override val dataModel =
        ConnectorGeneratorDataModel(
            connectorDevelopment = connectorDevelopmentType
        )

    init {
        helpButton.isVisible = false
        setSize(1300, 900)
        buildSteps()
        registerWizardSteps()
        init()
    }

    override fun buildSteps() {

        wizardStepsList.clear()

        buildConnectionStepsList()
        buildObjectClassStepsList()
    }

    private fun buildConnectionStepsList() {

        val connection = wizardStepsList.addCategory("Connection")

        connection.addStep(
            InitialContinueConnectorDevelopmentStep(
                this,
                client,
                GenerateConnectorBadge.State.IN_PROGRESS,
                false
            )
        )

        connection.addStep(
            BaseUrlSpecificationStep(
                this,
                client,
                GenerateConnectorBadge.State.NONE,
                false
            )
        )

        connection.addStep(
            AuthMethodSupportStep(
                this,
                client,
                GenerateConnectorBadge.State.NONE,
                false
            )
        )

        connection.addStep(
            AuthScriptsConnectorStep(
                this,
                client,
                GenerateConnectorBadge.State.NONE,
                false
            )
        )

        connection.addStep(
            CredentialsConnectorStep(
                this,
                client,
                GenerateConnectorBadge.State.NONE,
                false
            )
        )

        connection.addStep(
            TestConnectionStep(
                this,
                client,
                GenerateConnectorBadge.State.NONE,
                false
            )
        )
    }

    private fun buildObjectClassStepsList() {

        val connection = wizardStepsList.addCategory("Object Class")

        connection.addStep(
            InitialContinueConnectorDevelopmentStep(
                this,
                client,
                GenerateConnectorBadge.State.NONE,
                false
            )
        )
    }

    override fun doOKAction() {
        super.doOKAction()
    }
}